mlibrary/pride
# Reconfigure `esbuild`

# Overview
We officially say "Goodbye" to the original `./source` and welcome the new `./src` directory. `esbuild` now builds directly from `./src/index.js` and `spec/index.js` without needing the `concat.js` file to help assist the build. With `concat.js` gone, the `glob` dependency is no longer needed and has been uninstalled. Other changes include:

- Documentation relating to these changes have been updated, including pull request templates. 
- Everything `spec` related now lives under `./spec`.
- The package now points to `pride.min.js` for a smaller file size.

## Testing
1. Build the repository (`npm run build`).
2. Open `./spec/index.html` in the browser to see if all tests pass.
   * Break the new/updated unit tests to make sure they are working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#reconfigure-esbuild` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#reconfigure-esbuild"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
